subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/execution/SkipCachedTaskExecuterTest.groovy

```diff
@@ -244,18 +212,14 @@ class SkipCachedTaskExecuterTest extends Specification {
         ex.cause instanceof RuntimeException
         ex.cause.message == "Bad cacheIf() clause"
```

Comment:
We could also report the description of the `cacheIf()` clause here, if there's any. That would help debugging a lot. Not sure how frequently this would be used, though.